lib/em-hiredis/req_resp_connection.rb

```diff
@@ -0,0 +1,77 @@
+
+module EventMachine::Hiredis
+  module ReqRespConnection
+    include EventMachine::Hiredis::EventEmitter
+
+    def initialize
+      super
+      @response_queue = []
+      @connected = false
+    end
+
+    def send_command(df, command, args)
+      @response_queue.push(df)
+      puts "send #{command} #{args}"
+      send_data(marshal(command, *args))
+      return df
+    end
+
+    # EM::Connection callback
+    def connection_completed
+      puts "Connection completed"
+      @reader = ::Hiredis::Reader.new
+      @connected = true
+      emit(:connected)
+    end
+
+    # EM::Connection callback
+    def receive_data(data)
+      @reader.feed(data)
+      until (reply = @reader.gets) == false
```

Comment:
Left it as it was before. I'm too afraid of ruby truthiness to change it to anything other than `while (reply = @reader.gets) != false`.